### PR TITLE
Feat/hover-and-clicked-states

### PR DESCRIPTION
**Pull request by ValGeorgiev**
Hover and clicked state for share and save component icons

### CODE REVIEW

**Comment by nathggns** [comment]
Reviewing now.

File: packages/save-and-share-bar/src/save-and-share-bar.js [hunk 1]
```diff
@@ -13,64 +13,216 @@ import SharingApiUrls from "./constants";
...
+      twClicked,
+      twColour
+    } = this.state;
+
+    return (
+      <View style={styles.container}>
+        <View style={styles.rowItem}>
+          <Text style={styles.label}>Share</Text>
+          <View
+            onMouseEnter={() =>
```
[the remark this answers is not in the document]
Comment (nathggns) [reply packages/save-and-share-bar/src/save-and-share-bar.js]: If you hover over the Twitter icon, you'll see what I mean.

[the remark this answers is not in the document]
Comment (nathggns) [reply packages/save-and-share-bar/src/save-and-share-bar.js]: Just updated this a bit. https://codesandbox.io/s/vpkyz5vz7

[the remark this answers is not in the document]
Comment (ValGeorgiev) [reply packages/save-and-share-bar/src/save-and-share-bar.js]: Done, thanks

File: packages/save-and-share-bar/src/save-and-share-bar.js [hunk 2]
```diff
@@ -13,64 +14,178 @@ import SharingApiUrls from "./constants";
...
+              <IconEmail
+                fillColour="currentColor"
+                height={styles.svgIcon.height}
+                title="Share by email client"
+              />
+            </HoverIcon>
+          </Link>
+          <Link
+            onPress={this.onTwClick}
+            responsiveLinkStyles={twClicked ? styles.clickedLink : styles.link}
```
Comment (nathggns) [inline packages/save-and-share-bar/src/save-and-share-bar.js]: Have you considered extracting this clicked logic to another component so its not duplicated for each icon? Would make this component a lot simpler.

https://codesandbox.io/s/ym4r8kzzoj

Sorry for not mentioning it sooner, sometimes it's harder to see the possible simplifications when there's still a lot going on.

File: packages/save-and-share-bar/src/save-and-share-bar.js [hunk 3]
```diff
@@ -13,64 +14,178 @@ import SharingApiUrls from "./constants";
...
+              />
+            </HoverIcon>
+          </Link>
+        </View>
+      </View>
+    );
+  }
+}
+
+const HoverIcon = styled.div`
```
Comment (nathggns) [inline packages/save-and-share-bar/src/save-and-share-bar.js]: Can you define this before its used? Just a little bit easier to follow that way is all.

File: packages/save-and-share-bar/src/save-and-share-bar.js (hunk 2 above)
Comment (nathggns) [reply packages/save-and-share-bar/src/save-and-share-bar.js]: I think if you do this, the `SaveAndShareBar` can just become a stateless component.

Comment (nathggns) [reply packages/save-and-share-bar/src/save-and-share-bar.js]: It also might be worth just straight up adding this functionality to the Link package.

Comment (tuncaulubilge) [reply packages/save-and-share-bar/src/save-and-share-bar.js]: Adding hover support to link package is a good idea 👍

File: packages/save-and-share-bar/src/save-and-share-bar.js [hunk 4]
```diff
@@ -13,64 +14,178 @@ import SharingApiUrls from "./constants";
...
+            url={`${SharingApiUrls.twitter}?text=${articleUrl}`}
+          >
+            <HoverIcon
+              color={styles.svgIcon.fillColour}
+              hoverColor={styles.svgIcon.hoverFillColour}
+            >
+              <IconTwitter
+                fillColour="currentColor"
+                height={styles.svgIcon.height}
+                title="Share on tweeter"
```
Comment (nathggns) [inline packages/save-and-share-bar/src/save-and-share-bar.js]: Typo – tweeter.

File: packages/save-and-share-bar/src/save-and-share-bar.js (hunk 2 above)
Comment (nathggns) [reply packages/save-and-share-bar/src/save-and-share-bar.js]: This should give you an idea for how simple its possible to make this component https://gist.github.com/nathggns/f33f0fd4a7fad5e0401988f16b1460da

File: packages/save-and-share-bar/src/save-and-share-bar.js [hunk 5]
```diff
@@ -13,64 +14,178 @@ import SharingApiUrls from "./constants";
...
+            }
+          >
+            <HoverIcon
+              color={styles.svgIcon.fillColour}
+              hoverColor={styles.svgIcon.hoverFillColour}
+            >
+              <IconCopyLink
+                fillColour="currentColor"
+                height={styles.svgIcon.height}
+                title="Copy link on clipboard"
```
Comment (nathggns) [inline packages/save-and-share-bar/src/save-and-share-bar.js]: Should probably be `Copy link to clipboard`

File: packages/save-and-share-bar/src/save-and-share-bar.js [hunk 6]
```diff
@@ -13,64 +14,178 @@ import SharingApiUrls from "./constants";
...
+            url={`${SharingApiUrls.facebook}?text=${articleUrl}`}
+          >
+            <HoverIcon
+              color={styles.svgIcon.fillColour}
+              hoverColor={styles.svgIcon.hoverFillColour}
+            >
+              <IconFacebook
+                fillColour="currentColor"
+                height={styles.svgIcon.fb.height}
+                title="Share on facebook"
```
Comment (nathggns) [inline packages/save-and-share-bar/src/save-and-share-bar.js]: `Share on Facebook`

File: packages/save-and-share-bar/src/save-and-share-bar.js (hunk 4 above)
Comment (nathggns) [reply packages/save-and-share-bar/src/save-and-share-bar.js]: Should be `Share on Twitter`

**Comment by ValGeorgiev** [comment]
closing because something weird happen with style-components packages, will create a new one

**Pull request closed without merging**